src/vivarium/framework/values.py

```diff
@@ -170,6 +172,29 @@ def union_post_processor(values: List[NumberLike], _) -> NumberLike:
     return joint_value
 
 
+def step_size_post_processor(values: List[NumberLike], _) -> pd.Series:
+    """Computes the largest feasible step size for each simulant. This is the smallest component-modified
+    step size, or the global step size, whichever is larger. If no components modify the step size, we default
+    to the global step size.
+
+    Parameters
+    ----------
+    values
+        A list of step sizes
+
+    Returns
+    -------
+    pandas.Series
+        The largest feasible step size for each simulant
+
+
+    """
+    if len(values) == 1:
+        return values[0]
+    min_modified = pd.DataFrame(values[1:]).min(axis=0).astype("timedelta64[ns]")
+    return pd.DataFrame([values[0], min_modified]).max(axis=0)
```

Comment:
I'm kinda confused why you want the max of the two. It seems like it would be the min?

Reply:
In the current formulation, we don't actually want a local step that is smaller than the global step. So we're taking the smallest of the modifications, and taking the larger between that and the global step. This may possibly change a bit when we use the simulant step sizes themselves to modify the global step (MIC-4538)